test/EndToEnd/GivenAspNetAppsResolveImplicitVersions.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.DotNet.TestFramework;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using NuGet.ProjectModel;
+using NuGet.Versioning;
+using Xunit;
+
+namespace EndToEnd
+{
+    public class GivenAspNetAppsResolveImplicitVersions : TestBase
+    {
+        private const string LatestMinorVersion = "2.1";
```

Comment:
I'm going to look for a better way to determine the latest minor version.

Reply:
I would recommend patterning the tests more closely after what is here: https://github.com/dotnet/cli/blob/8c937a0db08e56660aca456ac088f2d0e70735ab/test/EndToEnd/GivenSelfContainedAppsRollForward.cs

- Version coverage
  - Have a list of versions to validate (right now this will only be 2.1, but should grow over time)
  - Have a test that creates a new project (via `dotnet new mvc`) and looks at the TargetFramework to determine what the latest major.minor version is, and then makes sure that version is in the list of versions to validate.
- Validating that the latest version is used
  - For this I think you need to restore a project that has the `Version` of a `PackageReference` to `Microsoft.AspNetCore.App` that floats (ie `2.1.*`), and then validate that the restored floating version matches the version you get implicitly.